### PR TITLE
Add requirements file for API autodocs on readthedocs

### PR DESCRIPTION
**Pull request by owlas**
Following up on #2314 

API docs now build on rtd. Here's my local version: http://pymc3-testing.readthedocs.io/en/rtd-docs/index.html

### CODE REVIEW

**Comment by kyleabeauchamp** [comment]
Sweet.  It would definitely be good to have the version drop-down menu to allow viewing both latest master and release docs, as I think we have a number of folks using features from master.

**Comment by owlas** [comment]
@twiecki - thanks for spotting the broken API docs. Errors in autodocs apparently do not show as build fails on RTD. Something to remember for future!

You shouldn't have to build docs locally anymore. Just ensure that any extra dependencies needed by PyMC3 are added to the `docs/requirements.txt` file.

**Comment by twiecki** [comment]
@owlas what's the fix?

**Comment by twiecki** [comment]
it's not possible to have a separate requirements file?

**Comment by ColCarroll** [comment]
@owlas any chance to use the `requirements.txt` and `requirements-dev.txt` in the top level directories?  If you think separate sphinx requirements are better, then maybe _remove_ the sphinx related requirements from `requirements-dev.txt`

**Comment by owlas** [comment]
Ah! I didn't look in `requirements-dev.txt` only in `requirements.txt`, which doesn't have nbsphinx needed to build the docs. I'll rebuild with `requirements-dev.txt`.

**Comment by twiecki** [comment]
You can leave `requirements.txt` as is then, right?

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+nbsphinx
```
Comment (twiecki) [inline docs/requirements.txt:1]: this shouldn't show up in the diff.

**Comment by twiecki** [comment]
@owlas I fixed it after some experimentation, thanks for your help!

**Pull request closed without merging**